tests/test_contact_specific_functions.py

```diff
@@ -352,69 +263,51 @@ def test_calculate_contact_forces_self_rod():
 
     "Testing function with analytically verified values"
 
-    tol = 1e-5
-
-    "initializing rod parameters"
-    rod_position_collection = np.array(
-        [[1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14, 15]]
-    )
-    rod_radius_collection = np.array(
-        [5.12952263, 0.19493561, 0.72904683, 0.21676892, 0.18705351]
-    )
-    rod_length_collection = np.array(
-        [0.89050271, 0.7076813, 0.21078658, 0.95372826, 0.86037329]
+    rod = MockRod()
+    """Increasing rod elements and length to establish self contact in rod;
```

Comment:
I think a better approach of doing this is to have the rod in a U-shape and increase the radius to make sure that the elements on either side of the U contact each other.